### PR TITLE
feat(vertexai/genai): add SystemInstruction

### PR DESCRIPTION
**Pull request by jba**
Expose GenerateContentRequest.SystemInstruction.

As usual, put it on the Model, and use the Model field to construct
each request.

### CODE REVIEW

**Review by eliben — APPROVED** [review]
It'd be nice to have an example for this too.

Particularly because this is called "system prompt" everywhere else, and this may make it difficult for users to discover. For example, there's an issue on generative-ai-go asking for it: https://github.com/google/generative-ai-go/issues/31

**Comment by jba** [comment]
Example forthcoming.

**Comment by jba** [comment]
I added a test and an example that shows various config options.
But I'm marking this a draft because the feature doesn't seem to be implemented.

**Comment by Deleplace** [comment]
> But I'm marking this a draft because the feature doesn't seem to be implemented.

The [Python sample](https://cloud.google.com/vertex-ai/generative-ai/docs/samples/generativeaionvertexai-gemini-system-instruction) for system instructions with Gemini 1.5 Pro is working for me. I'll see if can can get similar results with the Go sdk.

**Comment by Deleplace** [comment]
I'm getting compile errors at `example_test.go`, @jba can you PTAL?

File: vertexai/genai/example_test.go [hunk 1]
```diff
@@ -51,6 +51,31 @@ func ExampleGenerativeModel_GenerateContent() {
 	printResponse(resp)
 }
 
+// This example shows how to a configure a model. See [GenerationConfig]
+// for the complete set of configuration options.
+func ExampleGenerativeModel_GenerateContent_config() {
+	ctx := context.Background()
+	client, err := genai.NewClient(ctx, option.WithAPIKey(os.Getenv("GEMINI_API_KEY")))
```
Comment (eliben) [inline vertexai/genai/example_test.go:58]: Shouldn't this be with project/location and not key?

**Comment by Deleplace** [comment]
> > I'm getting compile errors at `example_test.go`, @jba can you PTAL?
> 
> Is it a vet error? What does it say?

```
% go test     
# cloud.google.com/go/vertexai/genai_test [cloud.google.com/go/vertexai/genai.test]
./example_test.go:58:38: undefined: option
./example_test.go:58:38: not enough arguments in call to genai.NewClient
	have (context.Context, unknown type)
	want (context.Context, string, string, ...option.ClientOption)
./example_test.go:58:56: undefined: os
FAIL	cloud.google.com/go/vertexai/genai [build failed]
```

And after fixing the imports:
```
./example_test.go:60:38: not enough arguments in call to genai.NewClient
```

The errors are trivial but I suspect we don't have jba's latest code.

**Comment by jba** [comment]
Sorry for the wait. I fixed the compilation problem and the feature now works with gemini-1.0-pro. Removing draft status.

File: vertexai/genai/example_test.go (hunk 1 above)
Comment (jba) [reply vertexai/genai/example_test.go]: Yes, fixed.

**Pull request merged**